Gems/Terrain/Assets/Shaders/Terrain/TerrainDetailHelpers.azsli

```diff
@@ -321,6 +389,8 @@ bool GetDetailSurface(inout DetailSurface surface, float2 detailMaterialIdCoord,
     TerrainSrg::m_detailMaterialIdImage.GetDimensions(textureSize.x, textureSize.y);
     int2 intTextureSize = textureSize;
 
+    surface = (DetailSurface)0;
```

Comment:
We have GetDefaultDetailSurface, might be a better fit.

Reply:
This actually needs to be zeroed out. Later, specific detail materials will be add their data to the surface, so you need to start at 0 for that to work correctly.